### PR TITLE
Add explainers on time bucketing

### LINKED ISSUE #910: [Content Bug] Lack of clarity about supported parameters in time_bucket within caggs and other time_bucket and cagg issues
# Describe the bug

Following the example results in an error: 

```
ERROR:  only immutable expressions allowed in time bucket function
HINT:  Use an immutable expression as third argument to the time bucket function.
```

## What do the docs say now?

Please see the discussion in https://github.com/timescale/timescaledb/issues/4117 as I am not 100% sure of the correction and where the corrections need to be made. Specifically this comment: https://github.com/timescale/timescaledb/issues/4117#issuecomment-1047562009 

## What should the docs say?

# Page affected
https://docs.timescale.com/api/latest/hyperfunctions/time_bucket_ng/#timescaledb-experimental-time-bucket-ng

# Version affected

latest

# Subject matter expert (SME)

[If known, who is a good person to ask about this topic]

# Screenshots
Screenshot 2022-03-08 at 09 27 57

# Any further info

[Anything else you want to add, or further links]

### PR DESCRIPTION
**Pull request by charislam**
# Description

Add more explanatory material on time bucketing. Currently the documentation on time bucketing is conceptual information mixed in with the API reference. There are a few nuances to get your head around (especially as you get into origins and time zones), so we would benefit from a more conceptual/how-to guide.

SMEs @NunoFilipeSantos @afiskon 

# Version

Which documentation version does this PR apply to?

- [x] Latest (Default)
- [ ] Version 1.7
- [ ] Older [specify]

# Links

Fixes #910 
- [x] Check 910 for specific points to be included

### CODE REVIEW

**Comment by Loquacity** [comment]
This is a great start, thanks for taking the initiative on this one! 💖

File: timescaledb/how-to-guides/time-buckets/index.md [hunk 1]
```diff
@@ -44,7 +46,7 @@ which is a Friday. It doesn't start on April 20, which is the immediately
 preceding Monday. It starts on April 13, because you can get to April 13, 2020,
 by counting in two-week increments from January 3, 2000.
 
-<!-- insert time bucket origin diagram -->
+<!-- TODO: insert time bucket origin diagram -->
```
Comment (Loquacity) [inline timescaledb/how-to-guides/time-buckets/index.md:49]: I think Vale looks for FIXME, should we add TODO?

**Review by afiskon — APPROVED** [review]
Looks OK to me

File: timescaledb/how-to-guides/time-buckets/index.md (hunk 1 above)
Comment (charislam) [reply timescaledb/how-to-guides/time-buckets/index.md:49]: I'm fine either way! Personally I use TODO for "would be good to have this but if we don't get to it for months that'll be fine, shouldn't block merge" and FIXME for "this is crucial and should block merge."

So I learn towards not cluttering up Vale with TODO admonitions, but I will bow to group consensus :)

Comment (Loquacity) [reply timescaledb/how-to-guides/time-buckets/index.md:49]: nope, that makes perfect sense. thanks!

**Review by Loquacity — APPROVED** [review]

**Comment by charislam** [comment]
Thanks for the review @afiskon! I added more, so do you want to give it another look?

**Comment by afiskon** [comment]
@charislam ✅

**Pull request merged**